src/Arcus.EventGrid.Publishing/EventGridPublisherBuilder.cs

```diff
@@ -53,19 +58,52 @@ public static EventGridPublisherBuilder ForTopic(string topicEndpoint)
         {
             Guard.NotNullOrWhitespace(topicEndpoint, nameof(topicEndpoint), "The topic endpoint must not be empty and is required");
 
+            bool isValidUri = Uri.TryCreate(topicEndpoint, UriKind.RelativeOrAbsolute, out Uri result);
```

Comment:
Do we use `Uri.TryCreate` and throw the exception ourselves or do we use `new Uri` that throws the exception itself?

Reply:
Yeah, not sure what to do.

* If I use the `Guard` class, I would need to have the result if the guard validation succeeds (therefore `TryCreate`)
* If I don't use the `Guard` class, I should throw the exception myself.

So, I'm not sure what the guidelines says about this? Should I favor the `Guard` above all? Like don't throw exceptions yourself for argument validation?

Reply:
I'm wondering why we just don't do `new Uri(topicEndpoint)` and leave it up to the framework to consider it a valid URL or not?

Reply:
Could do, but then we don't control the exception message.
If that's OK for you I'll use `new Uri(topicEndpoint)`.

Reply:
I think that's ok personally, if we get reports that it's an issue then we can still provide our own but I prefer to use the framework as much as possible for these kind of things instead of building our own

Reply:
Ok!